### ISSUE TITLE
Possible ConfigMapping issue with sibling/nested maps

### ISSUE THREAD

**Issue by Philipp-Borchert-ISH** [smallrye/smallrye-config#681]
Hi there,

while creating a custom config mapping I've got some weird looking exceptions while reading the config from a YAML file. 

tl;dr I've created a reproducer here: https://github.com/Philipp-Borchert-ISH/smallrye-config/commit/e9107f79f0dc2c385a844bc3fa3f6de4309f5841

This is the mapping:

```java
    @ConfigMapping(prefix = "parent")
    public interface Parent {

        @WithName("goodchildren")
        Map<String, GrandChild> goodChildren();

        @WithName("badchildren")
        Map<String, GrandChild> badChildren();

        interface GrandChild {
            @WithName("name")
            String name();
            @WithName("attributes")
            Map<String, String> attributes();
        }
    }
```

YAML example looks like this:

```yaml

parent:
  goodchildren:
    child1:
      name: John
      attributes:
        somekey: somevalue
        anotherkey: anothervalue
    child2:
      name: James
      attributes:
        something: isbroken
  badchildren:
    child3:
      name: BadJohn
      attributes:
        somekeybad: somevaluebad
        anotherkeybad: anothervaluebad 
```

Trying to parse the YAML results in the following exception:
```
java.lang.IllegalStateException: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	java.util.NoSuchElementException: SRCFG00014: The config property parent.badchildren.child3.attributes.name is required but it could not be found in any config source
	java.util.NoSuchElementException: SRCFG00014: The config property parent.goodchildren.child1.attributes.name is required but it could not be found in any config source
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:423)
	at io.smallrye.config.source.yaml.YamlConfigMappingSiblingTest.yamlConfigMapping(YamlConfigMappingSiblingTest.java:22
```

Apparently the model assumes the "name" attribute has to be in the "attributes" Map, although it is defined as a sibling of that Map. When you move the name attribute into the "attributes" block (in YAML) there won't be any exception.

Looks like this might be related to #520 / #630 - but I'm not sure, maybe I did something wrong with the Model definitions but right now I'm a bit lost ;-)

**Comment by Philipp-Borchert-ISH** [comment]
... I don't know why I didn't see this in the past 2 hours searching through issues here - could this be related to / duplicate of #667
?

**Comment by radcortez** [comment]
Let me check. This is a different case than the one in #667.

Actually, I believe this should be supported.

**Comment by radcortez** [comment]
#682 should fix this. Do you want to give it a try first?

**Comment by Philipp-Borchert-ISH** [comment]
Hi @radcortez, I see you've added the unit test to your PR so I'm confident this will be fine - thx a lot!
